packages/radio/index.js

```diff
@@ -0,0 +1,161 @@
+  foundation_ = null;
+  radioElement_ = React.createRef();
+  rippleActivator = null;
+
+  state = {
+    classList: new Set(),
+    disabled: false,
+    nativeControlId: '',
+  };
+
+  constructor(props) {
+    super(props);
+    this.foundation_ = new MDCRadioFoundation(this.adapter);
+  }
+
+  componentDidMount() {
+    this.foundation_.init();
+    const childProps = this.props.children.props;
+    if (childProps.disabled) {
+      this.foundation_.setDisabled(childProps.disabled);
+    }
+    if (childProps.id) {
+      this.setState({nativeControlId: childProps.id});
+    }
+    if (this.rippleActivator) {
+      this.props.initRipple(this.radioElement_.current, this.rippleActivator);
+    }
+  }
+
+  componentWillUnmount() {
+    this.foundation_.destroy();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const childProps = this.props.children.props;
+    if (childProps.disabled !== prevProps.children.props.disabled) {
+      this.foundation_.setDisabled(childProps.disabled);
+    }
+    if (childProps.id !== prevProps.children.props.id) {
+      this.setState({nativeControlId: childProps.id});
+    }
+  }
+
+  get classes() {
+    const {classList} = this.state;
+    const {className} = this.props;
+    return classnames('mdc-radio', Array.from(classList), className);
+  }
+
+  get adapter() {
+    return {
+      addClass: (className) => {
+        const classList = new Set(this.state.classList);
+        classList.add(className);
+        this.setState({classList});
+      },
+      removeClass: (className) => {
+        const classList = new Set(this.state.classList);
+        classList.delete(className);
+        this.setState({classList});
+      },
+      setNativeControlDisabled: (disabled) => this.setState({disabled}),
+    };
+  }
+
+  init = (el) => {
+    this.props.initRipple(el, this.inputControl_.current);
+  }
+
+  setRippleActivator = (element) => this.rippleActivator = element;
+
+  render() {
+    const {nativeControlId} = this.state;
+    const {
+      /* eslint-disable no-unused-vars */
+      label,
+      initRipple,
+      unbounded,
+      className,
+      /* eslint-enable no-unused-vars */
+      wrapperClasses,
+      ...otherProps
+    } = this.props;
+
+    return (
+      <div className={classnames('mdc-form-field', wrapperClasses)}>
+        <div className={this.classes} ref={this.radioElement_} {...otherProps}>
+          {this.renderNativeControl()}
+          <div className='mdc-radio__background'>
+            <div className='mdc-radio__outer-circle'></div>
+            <div className='mdc-radio__inner-circle'></div>
+          </div>
+        </div>
+        {label ? <label htmlFor={nativeControlId}>{label}</label> : null}
+      </div>
+    );
+  }
+
+  renderNativeControl() {
+    const children = React.Children.only(this.props.children);
+    const updatedProps = Object.assign({}, children.props, {
+      disabled: this.state.disabled,
+      setRippleActivator: this.setRippleActivator,
```

Comment:
we can pass the rippleActivator as a ref directly to `NativeControl` (see comment above)